### PR TITLE
fix can't start with -it flag

### PR DESCRIPTION
**Pull request by ningmingxiao**
Signed-off-by: ningmingxiao <ning.mingxiao@zte.com.cn>

### CODE REVIEW

**Comment by ningmingxiao** [comment]
fix https://github.com/containerd/nerdctl/issues/492

**Comment by AkihiroSuda** [comment]
`docker start` doesn't support tty

```
$ docker start --help

Usage:  docker start [OPTIONS] CONTAINER [CONTAINER...]

Start one or more stopped containers

Options:
  -a, --attach               Attach STDOUT/STDERR and forward signals
      --detach-keys string   Override the key sequence for detaching a container
  -i, --interactive          Attach container's STDIN
```

File: cmd/nerdctl/start.go [hunk 1]
```diff
@@ -95,13 +101,50 @@ func startContainer(ctx context.Context, container containerd.Container) error {
...
+		con = console.Current()
+		defer con.Reset()
+		if err := con.SetRaw(); err != nil {
+			return fmt.Errorf("failed to SetRaw,err is: %w", err)
+		}
+		if con == nil {
+			return errors.New("got nil con with flagT=true")
+		}
+		taskCIO = cio.NewCreator(cio.WithStreams(con, con, nil), cio.WithTerminal)
+	}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/start.go:119]: For consistency with Docker, we have to detach tty

Comment (ningmingxiao) [reply cmd/nerdctl/start.go:119]: what should I do?   change from   `return errors.New("got nil con with flagT=true")` to `return errors.New("got nil con")`?

Comment (AkihiroSuda) [reply cmd/nerdctl/start.go:119]: Probably you should attach a dummy terminal, not the real terminal, to emulate Docker's behavior.
Needs further investigation.

**Comment by fahedouch** [comment]
Fixed by https://github.com/containerd/nerdctl/pull/1642 ?

**Comment by ningmingxiao** [comment]
> Fixed by #1642 ? @ningmingxiao cc
yes

**Pull request closed without merging**